src/components/AddBook.js

```diff
@@ -1,17 +1,50 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/booksSlice';
 
-const AddBook = () => (
-  <div className="addBook">
-    <h3>Add New Book</h3>
-    <form className="addBookForm">
-      <input type="text" placeholder="Book Title" />
-      <select key="author">
-        <option value="Authors">Authors</option>
-      </select>
-      <button type="button">Add Book</button>
-    </form>
+const AddBook = () => {
+  const [values, setValue] = useState({});
+  const dispatch = useDispatch();
 
-  </div>
-);
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setValue((prevValues) => ({
+      ...prevValues,
+      [name]: value,
+      item_id: crypto.randomUUID(),
+    }));
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    dispatch(addBook(values));
+    setValue({});
+  };
+
+  return (
+    <section>
+      <h2>Add New Book</h2>
+      <form>
+        <input
+          value={values.title || ''}
+          type="text"
+          name="title"
+          placeholder="Book title"
+          onChange={handleChange}
+        />
+
+        <input
+          value={values.author || ''}
+          type="text"
+          name="author"
+          placeholder="author"
+          onChange={handleChange}
+        />
+
+        <button type="submit" onClick={handleSubmit}>Add Book</button>
+      </form>
+    </section>
```

Comment:
- [ ] [OPTIONAL] You can also add category as a drop down input field.
<img width="923" alt="253909755-cc4aa165-db0b-4d8a-9c20-ee6bf947db5f" src="https://github.com/AndreaM2429/Bookstore/assets/41163571/b83d4979-9255-4755-882b-5ebe57dc8db8">